packages/frontend/src/state/lp/hooks.ts

```diff
@@ -124,6 +125,59 @@ export const useClosePosition = () => {
   return closePosition
 }
 
+// Collect fees
+export const useCollectFees = () => {
+  const address = useAtomValue(addressAtom)
+  const controllerHelperContract = useAtomValue(controllerHelperHelperContractAtom)
+  const controllerContract = useAtomValue(controllerContractAtom)
+  const handleTransaction = useHandleTransaction()
+  const getDebtAmount = useGetDebtAmount()
+  const getVault = useGetVault()
+  const collectFees = useAppCallback(async (vaultId: number, onTxConfirmed?: () => void) => {
+    const vaultBefore = await getVault(vaultId)
+    const uniTokenId = vaultBefore?.NFTCollateralId    
+    
+    if (
+      !controllerContract ||
+      !controllerHelperContract ||
+      !address ||
+      !vaultBefore ||
+      !vaultBefore.shortAmount ||
+      !uniTokenId
+    )
+      return
+
+    const shortAmount = fromTokenAmount(vaultBefore.shortAmount, OSQUEETH_DECIMALS)
+    const debtInEth = await getDebtAmount(shortAmount)
+    const collateralToFlashloan = debtInEth.multipliedBy(COLLAT_RATIO)
```

Comment:
for all of these we need to have some buffer room.

If the transaction takes some time to be mined and spot prices change, its possible we won't be borrowing enough ETH.

Maybe 200% is fine?